packages/wp-source/src/actions.ts

```diff
@@ -29,20 +29,34 @@ const actions: WpSource["actions"]["source"] = {
     // Get options
     const force = options ? options.force : false;
 
-    if (!data || force) {
-      // Add the attributes that should be present even if fetch fails or we throw a ServerError below
+    if (!data) {
+      // If there is no data yet, just set the necessary flags:
       source.data[link] = {
-        link,
-        route: linkParams.route,
-        query,
-        page,
         isFetching: true,
         isReady: false,
       };
-    } else if (data.isReady || data.isFetching || data.isError) {
+    } else if (force) {
+      // If we fetch with `{ force: true }`, then wipe out whatever
+      // other data was present before and just set those flags:
+      source.data[link] = {
```

Comment:
This is wrong, we shouldn't wipe out the data, only set `data.isFetching` to true. Please read the issue in detail: https://github.com/frontity/frontity/issues/340

Reply:
This was my thought initially as well, but then I realized that if I implement this part like:

```
 source.data[link] = {
        ...source.data[link],
        isFetching: true,
        isReady: true,
      } as any;
```

Then we run into the following problem: If the previous `fetch()` has returned an error, the data for the error would not be removed. We would have to add additional logic to remove the `isError`, `errorStatusText` and `errorStatus`. 

This snapshot failure shows it:
```
  ● fetch › should run again when `force` is used

    expect(received).toMatchSnapshot()

    Snapshot name: `fetch should run again when \`force\` is used 1`

    - Snapshot
    + Received

    @@ -1,8 +1,11 @@
      Object {
        "/some/route/": Object {
    +     "errorStatus": 408,
    +     "errorStatusText": "Request Timeout",
          "id": 1,
    +     "isError": true,
          "isFetching": false,
          "isPostType": true,
          "isReady": true,
          "link": "/some/route/",
          "page": 1,

      138 |     await store.actions.source.fetch("/some/route/", { force: true });
      139 |     expect(handler.func).toHaveBeenCalled();
    > 140 |     expect(store.state.source.data).toMatchSnapshot();
          |                                     ^
      141 |   });
```

Why would it be incorrect to clear all the existing data if we run with `{force: true}`?

Reply:
> Why would it be incorrect to clear all the existing data if we run with {force: true}?

Because this is just a refresh of the content. The page needs to remain as it is.

Imagine you are on your twitter feed and click a "Refresh" button. You don't want your current feed to disappear, you just want that when the "Refresh" fetching ends, you see the new tweets appear in your feed.